lib/commands/push.ts

```diff
@@ -347,36 +357,46 @@ export default class PushCmd extends Command {
 			releaseTagValues.push('');
 		}
 
-		const app = appOrDevice;
 		await Command.checkLoggedIn();
-		const [token, baseUrl, owner] = await Promise.all([
+		const [token, baseUrl] = await Promise.all([
 			sdk.auth.getToken(),
 			sdk.settings.get('balenaUrl'),
-			this.getAppOwner(sdk, app),
 		]);
 
+		const application = (await getApplication(sdk, appNameOrSlug, {
+			$expand: {
+				organization: {
+					$select: ['handle'],
+				},
+			},
+			$select: ['app_name'],
+		})) as Application & {
+			organization: [Organization];
+		};
```

Comment:
This will throw `BalenaAmbiguousApplication` if user specifies an app name that matches multiple apps, which is the way we are handling this in other command, and which is I think the right way to do it.  In these cases app slugs should be used.